Run `python3 dev/update_changelog.py --prev-version 2.5...` (#9321)
mlflow/mlflow: CHANGELOG.md

# CHANGELOG

## 2.6.0 (2023-08-15)

MLflow 2.6.0 includes several major features and improvements

Features:

> - [Tracking] Supporting kwargs in log figure (#9179, @stroblme)

Reviewer: ```suggestion
- [Tracking] Supporting `save_kwargs` for `mlflow.log_figure` (#9179, @stroblme)
```

- [Gateway] Add support for MLflow serving to MLflow AI Gateway (#9199, @BenWilson2)

> - [Artifacts] Add progress bar (#9195, @serena-ruan)

Reviewer: ```suggestion
- [Artifacts] Display progress bars when uploading/download artifacts (#9195, @serena-ruan)
```

- [Models] Support logging the retriever model (#8808, @liangz1)
- [Tracking] Add support to log customized tags to runs created by autolog (#9114, @thinkall)

> - [Models / Scoring] Inference params support (#9068, @serena-ruan)

Reviewer: I think this entry deserves a better title :)

Bug fixes:

> - [] Enable F402 (#9305, @KekmaTime)

Reviewer: This isn't a bug fix.

- [Models] Fix text_pair functionality for transformers TextClassification pipelines (#9215, @BenWilson2)
- [Models] Fix LangChain compatibility with SQLDatabase (#9192, @dbczumar)
- [Tracking] Fix sklearn flavor `_get_metric_name_list` function (#9095, @WeichenXu123)

Documentation updates:

> - [Models] Mark mleap as deprecated (#9311, @BenWilson2)

Reviewer: I think this entry deserves a separate section.

Reviewer: We should also mention it'll be removed in a future release.

> - [Docs / Examples] Add examples and documentation for MLflow AI Gateway support for MLflow model serving (#9281, @BenWilson2)

Reviewer: `support for MLflow model serving` sounds like a new feature.

Reviewer: Ah the features section already has an entry for that.

# CHANGELOG

## 2.6.0 (2023-08-15)

MLflow 2.6.0 includes several major features and improvements

Features:

- [Tracking] Support `save_kwargs` for `mlflow.log_figure` to specify extra options when saving a figure (#9179, @stroblme)
- [Gateway] Add support for MLflow serving to MLflow AI Gateway (#9199, @BenWilson2)
- [Artifacts] Display progress bars when uploading/download artifacts (#9195, @serena-ruan)

> - [Models] Support logging retriever models in langchain (#8808, @liangz1)

Reviewer: ```suggestion
- [Models] Add support for logging LangChain's retriever models (#8808, @liangz1)
```

- [Tracking] Add support to log customized tags to runs created by autologging (#9114, @thinkall)

> - [Models / Scoring] Add support for passing extra params during inference for PyFunc models (#9068, @serena-ruan)

Reviewer: Let's move this to the top.

Bug fixes:

> - [Models] Fix text_pair functionality for transformers TextClassification pipelines (#9215, @BenWilson2)

Reviewer: ```suggestion
- [Models] Fix `text_pair` functionality for transformers `TextClassification` pipelines (#9215, @BenWilson2)
```

- [Models] Fix LangChain compatibility with SQLDatabase (#9192, @dbczumar)

> - [Tracking] Fix sklearn flavor `_get_metric_name_list` function (#9095, @WeichenXu123)

Reviewer: ```suggestion
- [Tracking] Avoid patching `sklearn.metrics.get_scorer_names` in `mlflow.sklearn.autolog` to avoid duplicate logging (#9095, @WeichenXu123)
```

Documentation updates:

- [Docs / Examples] Add examples and documentation for MLflow AI Gateway support for MLflow model serving (#9281, @BenWilson2)

> - [Docs / Examples] Add sentence-transformers doc & example (#9047, @es94129)

Reviewer: ```suggestion
- [Docs / Examples] Add `sentence-transformers` doc & example (#9047, @es94129)
```

Deprecation:

> - [Models] Mark mleap as deprecated. The `mlflow.mleap` module will be removed in a future release. (#9311, @BenWilson2)

Reviewer 2: /ai Do you have any suggestions on the changelog's content?

> - [Models] Mark mleap as deprecated. The `mlflow.mleap` module will be removed in a future release. (#9311, @BenWilson2)

Reviewer: ```suggestion
- [Models] The `mlflow.mleap` module has been marked as deprecated and will be removed in a future release (#9311, @BenWilson2)
```
